File: Implementierung/Backend/src/main/java/edu/kit/informatik/model/userdata/interactions/Teacher.java
package edu.kit.informatik.model.userdata.interactions;

import edu.kit.informatik.model.User;

public class Teacher extends Participant {

    public Teacher(User user, String firstName, String lastName) {
        super(user, firstName, lastName);
    }

    @Override
    public boolean isStudent() {
        return false;
    }
}

File: Implementierung/frontend/paschi/src/model/userdata/interactions/Teacher.ts
import {Participant} from "@/model/userdata/interactions/Participant";
import {User} from "@/model/User";

export class Teacher extends Participant {

  constructor(id: string | undefined, localId: number, user: User, firstName: string, lastName: string) {
    super(id, localId, user, firstName, lastName);
  }

  isTeacher(): boolean {
    return true;
  }
}
